File: playground/javascript/stack.js
class Node {
  constructor(value, next) {
    this.value = value;
    this.next = next;
  }
}

class Stack {
  constructor() {
    this.top = null;
  }

  pop() {
    if (this.top) {
      let topNode = this.top;
      this.top = this.top.next;
      return topNode.value;
    }
  }

  push(value) {
    let currTopNode = this.top;
    let newTopNode = new Node(value);
    newTopNode.next = currTopNode;
    this.top = newTopNode;
  }

  isEmpty() {
    return this.top === null;
  }
}

const stack = new Stack();

console.log(stack.isEmpty());
stack.push(1);
stack.push(2);
stack.push(3);
console.log(stack.isEmpty());
console.log(stack.pop());
console.log(stack.pop());
console.log(stack.pop());
console.log(stack.isEmpty());
console.log(stack.pop());

File: playground/dataStructure/stack.py
class Node:
    def __init__(self, value, next=None):
        self.value = value
        self.next = next
    # [curr]
    # [next]
    # [next]


class Stack:
    def __init__(self, value):
        self.top = Node(value)
        self.size = 0

    def push(self, value):
        new_node = Node(value)

        if self.top:
            new_node.next = self.top
        self.top = new_node
        self.size += 1

    def pop(self):
        curr = self.top

        while curr.next is not None:
            curr = curr.next

    def get_top(self):
        if self.top is not None:
            return self.top.value

    def is_empty(self):
        pass

    def clear(self):
        pass

    def print_all(self):
        curr = self.top
        stack = []
        while curr is not None:
            stack.append(curr.value)
            curr = curr.next

        print(stack)


a = Stack(1)
print(a.get_top())
a.push(2)
a.push(3)
a.push(4)
a.push(5)

a.print_all()
